cura/OAuth2/AuthorizationService.py

```diff
@@ -31,7 +31,10 @@ class AuthorizationService:
     account information.
     """
 
-    def __init__(self, settings: "OAuth2Settings", preferences: Optional["Preferences"] = None) -> None:
+    def __init__(self,
+                 settings: "OAuth2Settings",
+                 preferences: Optional["Preferences"] = None,
+                 get_user_profile: bool = True) -> None:
```

Comment:
I'd prefer if this variable wasn't a boolean, but a callable, defaulted to `self._getUserProfile`, which gives a lot more versatility against almost no cost. If it's not needed, just supply an empty method instead.

(Note that the optional part of the signature of the 'getUserProfile' function doesn't matter much, since when we need the proposed callback, it's called without arguments (other than self).)

Reply:
I did my best but it is not possible to have `self.getUserProfile` as default value because `self` is not defined at this point. So giving `None` gives you the default value and otherwise you specify your callback.

Reply:
Alright, then I think we can just merge this.